src/line_cursor.ts

```diff
@@ -486,6 +504,82 @@ export class LineCursor extends Marker {
     return this.getRightMostChild(newNode);
   }
 
+  /**
+   * Prepare for the deletion of a block by making a list of nodes we
+   * could move the cursor to afterwards and save it to
+   * this.potentialNodes.
+   *
+   * After the deletion has occurred, call postDelete to move it to
+   * the first valid node on that list.
+   *
+   * The locations to try (in order of preference) are:
+   *
+   * - The current location.
+   * - The connection to which the deleted block is attached.
+   * - The block connected to the next connection of the deleted block.
+   * - The parent block of the deleted block.
+   * - A location on the workspace beneath the deleted block.
+   *
+   * N.B.: When block is deleted, all of the blocks conneccted to that
+   * block's inputs are also deleted, but not blocks connected to its
+   * next connection.
+   *
+   * @param deletedBlock The block that is being deleted.
+   */
+  preDelete(deletedBlock: Blockly.Block) {
+    const curNode = this.getCurNode();
+
+    const nodes: Blockly.ASTNode[] = [curNode];
+    // The connection to which the deleted block is attached.
+    const parentConnection =
+      deletedBlock.previousConnection?.targetConnection ??
+      deletedBlock.outputConnection?.targetConnection;
+    if (parentConnection) {
+      const parentNode = Blockly.ASTNode.createConnectionNode(parentConnection);
+      if (parentNode) nodes.push(parentNode);
```

Comment:
(For team discussion, and not a blocker for this PR)

I strongly prefer to only use single-line `ifs` for returns. If there's anything at all happening in the if, such as here, I prefer to have the brackets. The style guide says they _may_ be elided; I believe in core we only elide them for returns.